sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py

```diff
@@ -124,7 +129,31 @@ def key_id(self):
 
         :rtype: str
         """
-        return self._key_id.source_id
+        if not self._jwk:
+            return self._key_id.source_id
+        return self._key_id
+
+    @classmethod
+    def from_jwk(cls, jwk):
+        # type: (Union[JsonWebKey, dict]) -> CryptographyClient
+        """Creates a client that can only perform cryptographic operations locally.
+
+        :param jwk: the key's cryptographic material, as a JsonWebKey or dictionary.
+        :type jwk: JsonWebKey or dict
+        :rtype: CryptographyClient
+
+        .. literalinclude:: ../tests/test_examples_crypto.py
+            :start-after: [START from_jwk]
+            :end-before: [END from_jwk]
+            :caption: Create a CryptographyClient from a JsonWebKey
+            :language: python
+            :dedent: 8
+        """
+        if isinstance(jwk, JsonWebKey):
+            key = vars(jwk)
+        else:
+            key = jwk
+        return cls(key, object(), _jwk=True)
```

Comment:
This goes against the prescribed type for the key parameter, but I feel like that's reasonable for the implementation